plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -0,0 +1,408 @@
+            elif unit_flags == 'GT':
+                rel_flags = solv.REL_GT
+            elif unit_flags == 'LE':
+                rel_flags = solv.REL_EQ | solv.REL_LT
+            elif unit_flags == 'GE':
+                rel_flags = solv.REL_EQ | solv.REL_GT
+            else:
+                raise ValueError('Unsupported dependency flags %s' % unit_flags)
+            dep = dep.Rel(rel_flags, pool.Dep(unit_evr))
+        # register the constructed solvable dependency
+        solvable.add_deparray(keyname, dep)
+
+
+def rpm_dependency_attribute_factory(attribute_name, dependency_key=None):
+    return repeated_attr_conversion(attr_conversion(attribute_name, default=[])())(
+        lambda solvable, unit: rpm_dependency_conversion(
+            solvable, unit, attribute_name, dependency_key=dependency_key
+        ))
+
+
+def rpm_filelist_conversion(solvable, unit):
+    """A specific, rpm-unit-type filelist attribute conversion."""
+    repodata = solvable.repo.first_repodata()
+    unit_files = unit.get('files', {}).get('file', [])
+    unit_files.extend(unit.get('files', {}).get('dir', []))
+    for filename in unit_files:
+        dirname = os.path.dirname(filename).encode('utf-8')
+        dirname_id = repodata.str2dir(dirname, create=True)
+        repodata.add_dirstr(solvable.id, solv.SOLVABLE_FILELIST,
+                            dirname_id, os.path.basename(filename).encode('utf-8'))
+
+
+def unit_solvable_converter(solv_repo, unit, *attribute_factories):
+    """Create a factory of a content unit--solv.Solvable converter.
+
+    Each attribute factory either calls setattr on a solvable with a converted attribute value
+    or processes the attribute value and changes the state of either the solvable being created
+    or the solv.Repo or both by e.g adding dependencies.
+
+    :param attribute_factories: the attribute factories to use for the conversion
+    :type attribute_factories: a list of (solvable, unit, repo=None) -> None callables
+    :param solv_repo: the repository the unit is being added into
+    :type solv_repo: solv.Repo
+    :param unit: the unit being converted
+    :type unit: pulp_rpm.plugins.models.Model
+    :return: the solvable created.
+    :rtype: solv.Solvable
+    """
+    solvable = solv_repo.add_solvable()
+    for attribute_factory in attribute_factories:
+        attribute_factory(solvable, unit)
+    return solvable
+
+
+def unit_solvable_converter_factory(*attribute_factories):
+    return lambda solv_repo, unit: unit_solvable_converter(solv_repo, unit, *attribute_factories)
+
+
+rpm_unit_solvable_factory = unit_solvable_converter_factory(
+    # An RPM content unit nests dependencies in a dict format
+    # Would be provided by pulp_rpm
+    plain_attribute_factory('name'),
+    evr_attribute,
+    plain_attribute_factory('arch'),
+    plain_attribute_factory('vendor'),
+    rpm_dependency_attribute_factory('requires'),
+    rpm_dependency_attribute_factory('conflicts'),
```

Comment:
I shall remove the unused fields for now; if in the future required, we'd better add on an as-needed basis.

Reply:
How come that conflicts do not exist?!

Reply:
we don't track these in Mongo so it has no effect; Can a repo have conflicting units inside? I guess it can, or?